lib/library.ml

```diff
@@ -0,0 +1,64 @@
+let rec collect_cmi_files dir =
+  let open CCResult.Infix in
+  let* files =
+    try Ok (Sys.readdir dir)
+    with Sys_error e ->
+      Error (Printf.sprintf "Error reading directory %s: %s" dir e)
+  in
+  Array.fold_left
+    (fun acc_res file ->
+      let* acc = acc_res in
+      let path = Filename.concat dir file in
+      try
+        if Sys.is_directory path then
```

Comment:
This should ignore directories and only load `.cmi` files from the given directory. The idea is that users should call this function on, for instance, `_build/install/default/lib/re` or `_opam/lib/re`.

I.e. let's keep this as simple as possible and not bother with lib discovery logic for now. The current implem is likely to load a lot of irrelevant modules.